coverage.py

```diff
@@ -95,6 +95,28 @@ def __init__(self, *args, **kwargs):
         # a single run of the CoverageProvider.
         self.offset = 0
 
+        self.successes = 0
+        self.transient_failures = 0
+        self.persistent_failures = 0
+
+    @property
+    def achievements(self):
+        """Represent the achievements of a CoverageProvider as a
+        human-readable string.
+        """
+        template = "Items processed: %d. Successes: %d, transient failures: %d, persistent failures: %d"
+        total = (self.successes + self.transient_failures
+                 + self.persistent_failures)
+        return template % (
+            total, self.successes, self.transient_failures,
+            self.persistent_failures
+        )
+
+    @achievements.setter
+    def achievements(self, value):
+        # It's not possible to set .achievements directly. Do nothing.
+        pass
```

Comment:
Wouldn't it be better to raise an error? Or is this needed because of TimestampData?

Reply:
Yeah, the TimestampData constructor sets .achievements to None.